### ISSUE TITLE
segmentation fault in test_avl.c

### ISSUE THREAD

**Issue by ClaudioDaffra** [jarun/dslib#5]
compiling the example with test_avl.c i get a segmentation fault

test/test_avl_1.c :

	claudio@xubuntu20vm:~/dslib$ ./x
	[lib/avl.c, search_BFS_avl(), ln 527] INFO: tracking...
	[lib/avl.c, search_BFS_avl(), ln 527] INFO: tracking...
	[lib/avl.c, search_BFS_avl(), ln 532] INFO: FOUND 10
	Errore di segmentazione (core dump creato)
	claudio@xubuntu20vm:~/dslib$

**Comment by jarun** [comment]
A more detailed report from address sanitizer.

Ran the tests under ASan, UBSan, and TSan:

```
cc -g -fsanitize=address,undefined -Iinclude src/*.c test/test_queue_1.c

ERROR: AddressSanitizer: stack-buffer-overflow
READ of size 4
    #0 0x558e591d44ad in main test/test_queue_1.c:52
```

The queue returns a garbage value, resulting in a stack overflow during the test.

```
cc -g -fsanitize=address,undefined -Iinclude src/*.c test/test_stack_1.c

ERROR: LeakSanitizer: detected memory leaks

Direct leak of 8 byte(s) in 1 object(s) allocated from:
    #1 0x55a7247d3f48 in get_stack src/stack.c:29
    #2 0x55a7247d72c6 in main test/test_stack_1.c:33

Indirect leak of 8 byte(s) in 1 object(s) allocated from:
    #1 0x55a7247d3f5b in get_stack src/stack.c:31
    #2 0x55a7247d72c6 in main test/test_stack_1.c:33

SUMMARY: AddressSanitizer: 16 byte(s) leaked in 2 allocation(s).
```

There's a memory leak in the stack. The first head element is somehow lost and never freed.

```
cc -g -fsanitize=address,undefined -Iinclude src/*.c test/test_avl_1.c

ERROR: AddressSanitizer: heap-use-after-free
READ of size 8
    #0 0x55d69df5ce7c in destroy_queue src/queue.c:82
    #1 0x55d69df58e65 in search_BFS_avl src/avl.c:573
    #2 0x55d69df607ee in main test/test_avl_1.c:50

freed by thread T0 here:
    #1 0x55d69df5cefb in destroy_queue src/queue.c:86
    #2 0x55d69df58708 in search_BFS_avl src/avl.c:533
    #3 0x55d69df607ee in main test/test_avl_1.c:50

previously allocated by thread T0 here:
    #1 0x55d69df5cb5e in get_queue src/queue.c:29
    #2 0x55d69df584a8 in search_BFS_avl src/avl.c:515
    #3 0x55d69df607ee in main test/test_avl_1.c:50
```

The AVL tree has a use-after-free defect. Queues are being destroyed twice.

**Comment by jarun** [comment]
First 2 fixed on master. Yet to check the 3rd one.

**Comment by jarun** [comment]
Fixed at commit 58fb92c24da07cda6ee661cb81398f55139602cd.
